zkevm-circuits/src/tx_circuit.rs

```diff
@@ -1675,6 +1675,17 @@ impl<F: Field> SubCircuit<F> for TxCircuit<F> {
             .collect::<Result<Vec<SignData>, Error>>()?;
 
         config.load_aux_tables(layouter)?;
+
+        // assert tx.caller_address == recovered_pk
+        for (sign_data, tx) in keccak_inputs_sign_verify(&sign_datas)
+            .into_iter()
+            .zip(self.txs.iter())
+        {
+            let pk_hash = keccak(&sign_data);
+            let address = pk_hash.to_address();
+            assert_eq!(address, tx.caller_address);
+        }
```

Comment:
suggest change to log::error! so the test tx_circuit_bad_address will not panic

Reply:
fixed with https://github.com/scroll-tech/zkevm-circuits/pull/345/commits/89fafcd8034338adf8a5945433adb36c944748c7